waspc/cli/Command/Common.hs

```diff
@@ -25,30 +26,44 @@ import System.Directory
   )
 import qualified System.FilePath as FP
 
-findWaspProjectRoot :: Path Abs (Dir ()) -> Command (Path Abs (Dir WaspProjectDir))
-findWaspProjectRoot currentDir = do
+findWaspProjectRootIO :: Path Abs (Dir ()) -> IO (Maybe (Path Abs (Dir WaspProjectDir)))
+findWaspProjectRootIO currentDir = do
   let absCurrentDirFp = SP.toFilePath currentDir
-  doesCurrentDirExist <- liftIO $ doesPathExist absCurrentDirFp
-  unless doesCurrentDirExist (throwError notFoundError)
-  let dotWaspRootFilePath = absCurrentDirFp FP.</> SP.toFilePath dotWaspRootFileInWaspProjectDir
-  isCurrentDirRoot <- liftIO $ doesFileExist dotWaspRootFilePath
-  if isCurrentDirRoot
-    then return $ SP.castDir currentDir
+  doesCurrentDirExist <- doesPathExist absCurrentDirFp
+  if (not doesCurrentDirExist)
+    then return Nothing
     else do
-      let parentDir = SP.parent currentDir
-      when (parentDir == currentDir) (throwError notFoundError)
-      findWaspProjectRoot parentDir
-  where
-    notFoundError =
-      CommandError
-        ( "Couldn't find wasp project root - make sure"
-            ++ " you are running this command from Wasp project."
-        )
-
-findWaspProjectRootDirFromCwd :: Command (Path Abs (Dir WaspProjectDir))
-findWaspProjectRootDirFromCwd = do
-  absCurrentDir <- liftIO getCurrentDirectory
-  findWaspProjectRoot (fromJust $ SP.parseAbsDir absCurrentDir)
+      let dotWaspRootFilePath = absCurrentDirFp FP.</> SP.toFilePath dotWaspRootFileInWaspProjectDir
+      dotWaspRootFileExists <- doesFileExist dotWaspRootFilePath
+      if dotWaspRootFileExists
+        then return $ Just (SP.castDir currentDir)
+        else do
+          let parentDir = SP.parent currentDir
+          if (parentDir == currentDir)
+            then return Nothing
+            else findWaspProjectRootIO parentDir
+
+findWaspProjectRootCmd :: Path Abs (Dir ()) -> Command (Path Abs (Dir WaspProjectDir))
```

Comment:
You added `Cmd` to two of the function in this file, which makes sense, it is a way to differ them from their `IO` versions.

However, since they are in `Command.Common`, I think that is actually not needed! And they are used on quite a few places, so having that `Cmd` suffix makes it harder to read them. So I would just remove that `Cmd` suffix, from both this and another function in this file.